scene/gui/code_edit.cpp

```diff
@@ -1398,6 +1402,319 @@ PackedInt32Array CodeEdit::get_executing_lines() const {
+			bool with_alpha = p_color.a < 1.0 || key.count(",") == 3;
+
+			with = "Color.from_ok_hsl(" + rtos(p_color.get_ok_hsl_h()) + ", " +
+					rtos(p_color.get_ok_hsl_s()) + ", " +
+					rtos(p_color.get_ok_hsl_l());
+			if (with_alpha) {
+				with += ", " + rtos(p_color.a);
+			}
+			with += ")";
+			break;
+		}
+		case ColorPreviewFormat::FN_FROM_RGBE9995: {
+			bool with_prefix = key.contains("0x");
+			String prefix = with_prefix ? "0x" : "";
+
+			with = "Color.from_rgbe9995(" + prefix + String::num_uint64(p_color.to_rgbe9995(), with_prefix ? 16 : 10) + ")";
+			break;
+		}
+		case ColorPreviewFormat::FN_HEX: {
+			bool with_prefix = key.contains("0x");
+			String prefix = with_prefix ? "0x" : "";
+
+			with = "Color.hex(" + prefix + String::num_uint64(p_color.to_rgba32(), with_prefix ? 16 : 10) + ")";
+			break;
+		}
+		case ColorPreviewFormat::FN_HTML: {
+			String with_prefix = key.contains("#") ? "#" : "";
+
+			with = "Color.html(\"" + with_prefix + p_color.to_html() + "\")";
+			break;
+		}
+		case ColorPreviewFormat::FN_COLOR8:
+		case ColorPreviewFormat::FN_COLOR8_ALPHA: {
+			bool with_alpha = p_color.a < 1.0 || key.count(",") == 3;
+
+			with = "Color8(" + itos(p_color.get_r8()) + ", " +
+					itos(p_color.get_g8()) + ", " +
+					itos(p_color.get_b8());
+			if (with_alpha) {
+				with += ", " + itos(p_color.get_a8());
+			}
+			with += ")";
+			break;
+		}
+		case ColorPreviewFormat::CTOR_VEC4_SOURCE_COLOR:
+			with = "vec4(" + rtos(p_color.r) + ", " +
+					rtos(p_color.g) + ", " +
+					rtos(p_color.b) + ", " +
+					rtos(p_color.a) + ")";
+			break;
+		case ColorPreviewFormat::TEXT_HEX_RGBA:
+		case ColorPreviewFormat::TEXT_HTML_RGBA: {
+			bool with_alpha = p_color.a < 1.0 || key.length() == 10;
+			int start = color_preview_match->get_start(0);
+			int end = color_preview_match->get_end(0);
+			int hex_start = color_preview_match->get_start("hex");
+			int hex_end = color_preview_match->get_end("hex");
+			String prefix = new_line.substr(start, hex_start - start);
+			String suffix = new_line.substr(hex_end, end - hex_end);
+
+			with = prefix + p_color.to_html(with_alpha) + suffix;
+			break;
+		}
+	}
+	new_line = new_line.replace(key, with);
+	begin_complex_operation();
+	set_line(p_line, new_line);
+	end_complex_operation();
+	queue_redraw();
+	// NOTE: update `match` and `format` with the new color. This is required
+	//       to keep ColorPicker and editor's lines in sync.
+	Color color;
+
+	color_preview_match = _parse_color_from_line(new_line, color, color_preview_format);
+}
+
+void CodeEdit::_color_preview_closed() {
+	color_preview_line = -1;
+}
+
+void CodeEdit::_color_preview_updated(const Color &p_color) {
+	_update_color_in_line(color_preview_line, p_color);
+}
+
+void CodeEdit::_color_preview_gutter_draw_callback(int p_line, int p_gutter, Rect2 p_region) {
+	String line = get_line(p_line);
+	Color color;
+	ColorPreviewFormat format;
+	Ref<RegExMatch> match = _parse_color_from_line(line, color, format);
```

Comment:
As it stands, this does a *lot* of string match calls, for each visible line, each time the CodeEdit is redrawn.
I think an initial regex which weeds out all unrelated lines that have nothing related to `Color` in them might make sense.